Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector_test.py

```diff
@@ -1,123 +1,152 @@
 import json
-import io
+from urllib.parse import urljoin
+import pytest
+from pytest_mock.plugin import MockerFixture
+from requests_mock.mocker import Mocker as RequestsMock
+from DigitalGuardianARCEventCollector import Client
+
+
+CLIENT_KWARGS = {
+    'verify': False,
+    'proxy': False,
+    'auth_url': 'https://example.com',
+    'base_url': 'https://example.com',
+    'client_id': '11',
+    'client_secret': '22',
+    'export_profile': '33',
+}
 
 
 def util_load_json(path):
-    with io.open(path, mode='r', encoding='utf-8') as f:
+    with open(path, encoding='utf-8') as f:
         return json.loads(f.read())
 
 
-def test_add_time_key_to_events():
+@pytest.fixture
+def authenticated_client(requests_mock: RequestsMock) -> Client:
+    """Fixture to create a Digital Guardian Client instance."""
+    token_url = urljoin(CLIENT_KWARGS['auth_url'], '/as/token.oauth2')
+    requests_mock.post(token_url, json={'access_token': '123', 'expires_in': 10000})
+
+    return Client(**CLIENT_KWARGS)
+
+
+def test_get_access_token(mocker: MockerFixture):
+    """
+    Given:
+        - A valid access token in the integration context
+    When:
+        - Calling Client._get_access_token
+    Then:
+        - Ensure the token in the integration context is returned and no new token is requested.
     """
-       Given:
-           - list of events
-       When:
-           - Calling add_time_key_to_events
-       Then:
-           - Ensure the _time key is added to the events
-       """
-    from DigitalGuardianARCEventCollector import add_time_to_events
+    integration_context_token = {'token': '123', 'valid_until': 2022222222}
+    mocker.patch('DigitalGuardianARCEventCollector.get_integration_context', return_value=integration_context_token)
+    mocker.patch('time.time', return_value=1022222222)
```

Comment:
instead of patching `time`, use `freeze_time`.

Reply:
Updated to use `freeze_time` in commit 582763c169821e56b0a063bc42e00239eb151efc